.github/workflows/ci.yml

```diff
@@ -113,16 +113,79 @@ jobs:
             -   name: Checkout
                 uses: actions/checkout@v3
 
+            -   name: Setup PHP to install dependencies
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: 5.5
+                    ini-values: error_reporting=-1, display_errors=On, zend.assertions=1
+                    coverage: none
+                    # Autoload files generated with Composer 2.3+ are not compatible with PHP < 7.0.
+                    tools: composer:2.2
+
+            -   name: Install Composer dependencies
+                uses: ramsey/composer-install@v2
+                with:
+                    composer-options: "--no-dev --optimize-autoloader"
+
+            -   name: Setup PHP to generate the PHAR
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: 8.1
+                    extensions: exif, phar, openssl, sodium
+                    coverage: none
+                    ini-values: phar.readonly=Off, error_reporting=-1, display_errors=On, zend.assertions=1
+                    # Autoload files generated with Composer 2.3+ are not compatible with PHP < 7.0.
+                    tools: composer:2.2
+
+            # Note: do NOT turn on the requirement checker in the box config as it is no longer
+            # compatible with PHP < 7.2.
+            -   name: Install Box
+                run: wget https://github.com/humbug/box/releases/latest/download/box.phar -O box.phar && chmod 0755 box.phar && pwd
+
+            -   name: Validate configuration
+                run: php box.phar validate -i box.json
+
+            -   name: Building binary...
+                run: php box.phar compile -v --config=box.json
+
+            -   name: Show info about the build phar
+                run: php box.phar info -l ./build/artifacts/php-coveralls.phar
+
+            -   name: Upload the Phar artifact
+                uses: actions/upload-artifact@v3
+                with:
+                    name: php-coveralls-phar
+                    path: ./build/artifacts/php-coveralls.phar
+                    retention-days: 30
```

Comment:
do you see it gonna be used anywhere outside of the CI? maybe we don't need 30 days here, like as soon as CI is green, we don't need this file stored for each commit job, I believe

Reply:
The reason I set this to `30` is that it will allow for testing PRs with the PHAR. Each PR build will generate the PHAR artifact, which can be downloaded from the GH Actions CI build for that PR and used to verify the PR. And as PRs have a tendency to stay open for a while in this repo, `30` days seemed like a good number.

Reply:
👍🏻 
I didn't considered that the artifact is exposed to outside than internal CI storage